### PR TITLE
Feat/undeletions

### LINKED ISSUE #1469: Clarify definition of mutation categories
Congrats on the new release!  It's exciting to see the new mutation categories, and I'll have to explode some trees and see what happens.  :)  I just noticed in an [UShER subtree view](https://nextstrain.org/fetch/hgwdev.gi.ucsc.edu/~angie/auspice_rev_homoplasy.json?branchLabel=back-mutations&c=pango_lineage_usher&label=nuc%20mutations:C2790T,C9866T,T22200G,G22599A,C24503T,T26060C) that the same mutation can be listed as both a Homoplasy and Reversion in the branch pop-up, which I wouldn't expect.  I don't know if it's my [JSON](https://hgwdev.gi.ucsc.edu/~angie/auspice_rev_homoplasy.json) or a logic corner-case or the expected behavior...?

**Current Behavior**  
The reversion G22200T is listed under both "Homoplasies" and "Reversions to reference" in the pop-up:
image

**Expected behavior**  
I would expect the categories to be mutually exclusive, so G22200T would be just a reversion, not both a reversion and a homoplasy.  I'm not really sure what homoplasy is intended to mean here... is it anything that happens more than once in the JSON, or is it more like Nextclade's new "labeledSubstitution" output?  Or something else?  :)

**How to reproduce**  

https://nextstrain.org/fetch/hgwdev.gi.ucsc.edu/~angie/auspice_rev_homoplasy.json?branchLabel=back-mutations&c=pango_lineage_usher&label=nuc%20mutations:C2790T,C9866T,T22200G,G22599A,C24503T,T26060C

Hover or shift-click on any labeled branch (branches are labeled by reversions).

### LINKED ISSUE #1537: ENH: Split up deletions and substitutions in tooltip as deletions are overwhelming and uninformative
### Context
When inspecting trees I often have issues with the way the presentation of deletions overwhelms the display.

### Description
Deletions are often very noisy on trees as we don't account for them properly when building a tree.

Yet they take up most space because they come in stretches that aren't compressed.

### Examples
So I end up with a result like this:

image

Where the needle (the two really important nuc substitutions) is really hard to find in the hay stack (long list of deletions).

It would be really great and quite high priority from my user perspective to make deletions less overwhelming.

This has actually impeded me for a while but I never had the idea of writing it up as an issue.

### Possible solution

@victorlin do you think this is something you could have a look at? There are a few things we could try here:
1. Simply split out substitutions and deletions (maybe easiest and quickest, maybe stopgap until we have)
2. Compress deletion stretches (and maybe separate them out, too)

### PR DESCRIPTION
**Pull request by jameshadfield**
Closes #1537, which has context on the rational for this PR.
Closes #1469

Undeletions (can we think of a better word than "undeletion"?!?) on branches are split out into a separate category and grouped into intervals, like so: 
image

This is not done for tips, as these don't report mutations but rather changes between the root and the tip. While undeletions can be present here, it's rarer and I don't think any of our core builds would have this. See the tests introduced in 9fa12980602327e5bf23a22ac981c244504e8345 for examples of this.

The third commit adds a info-popup detailing what the different categories are and how mutations are grouped into them.

image
image

### CODE REVIEW

**Comment by emmahodcroft** [comment]
We could say 'deletion reversion' rather than 'undeletion' - just depends on space I suppose!

One could extend the description of gaps to say something like "A change to a '-' character, indicating a missing base. These can indicate deletions but sometimes areas of no coverage are filled with gaps." - but I've no strong feelings and this may be too wordy.

Overall I think this looks good, I like the mouseover option to see things more clearly! Thanks James!

**Review by emmahodcroft — APPROVED** [review]

**Review by joverlee521 — APPROVED** [review]
Really happy to see the tooltip for the mutations! IIRC, we've had many questions regarding what the mutation labels mean.

File: src/components/tree/infoPanels/MutationTable.js [hunk 1]
```diff
@@ -131,6 +133,49 @@ const displayGeneMutations = (gene, mutsPerCat) => {
...
+  A summary of sequence changes between the root and the selected tip.
+  Changes are grouped into one of the following (mutually exclusive) categories,
+  with the first matching category used:
+
+  <ul>
+    <li>Gaps: A change to a &apos;-&apos; character, indicating a missing base; these can indicate deletions but sometimes areas of no coverage are filled with gaps</li>
+    <li>Ns: Typically due to lack of sequence coverage or ambiguity at this position (Nucleotides only)</li>
+    <li>Reversions to root: The tip state is the same as the root state, however this has changed and been reverted along the way</li>
+    <li>Changes: The tip state differs from the root state</li>
+  </ul>
```
[the remark this answers is not in the document]
Comment (jameshadfield) [reply src/components/tree/infoPanels/MutationTable.js]: Good idea -- changed & amended the commit
